operators/src/plot/histogram.rs

```diff
@@ -0,0 +1,841 @@
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedPlotOperator,
+    Operator, PlotOperator, PlotQueryProcessor, PlotResultDescriptor, QueryContext, QueryProcessor,
+    QueryRectangle, TypedPlotQueryProcessor, TypedRasterQueryProcessor, TypedVectorQueryProcessor,
+};
+use crate::error;
+use crate::error::Error;
+use crate::string_token;
+use crate::util::Result;
+use futures::future::BoxFuture;
+use futures::stream::BoxStream;
+use futures::{FutureExt, StreamExt, TryFutureExt};
+use geoengine_datatypes::collections::{FeatureCollection, FeatureCollectionInfos};
+use geoengine_datatypes::plots::{Plot, PlotData};
+use geoengine_datatypes::primitives::{
+    DataRef, FeatureDataRef, FeatureDataType, Geometry, Measurement,
+};
+use geoengine_datatypes::raster::{Pixel, RasterTile2D};
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, OptionExt};
+use std::convert::TryFrom;
+
+pub const HISTOGRAM_OPERATOR_NAME: &str = "Histogram";
+
+/// A histogram plot about either a raster or a vector input.
+///
+/// For vector inputs, it calculates the histogram on one of its attributes.
+///
+pub type Histogram = Operator<HistogramParams>;
+
+/// The parameter spec for `Histogram`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct HistogramParams {
+    /// Name of the (numeric) attribute to compute the histogram on. Ignored for operation on rasters.
+    pub column_name: Option<String>,
```

Comment:
is this info really necessary here? And if so, also the unit of the column would be interesting. Still it's a bit unelegant to have this field also in raster histograms.

Reply:
e91ddcc